kusto/ingest/status.go

```diff
@@ -0,0 +1,301 @@
+	Skipped StatusCode = "Skipped"
+	// PartiallySucceeded status represents a permanent status.
+	// Part of the data was successfully ingested to Kusto, while other parts failed.
+	PartiallySucceeded StatusCode = "PartiallySucceeded"
+
+	// StatusRetrievalFailed means the client ran into truble reading the status from the service
+	StatusRetrievalFailed StatusCode = "StatusRetrievalFailed"
+	// StatusRetrievalCanceled means the user canceld the status check
+	StatusRetrievalCanceled StatusCode = "StatusRetrievalCanceled"
+)
+
+// IsFinal returns true if the ingestion status is a final status, or false if the status is temporary
+func (i StatusCode) IsFinal() bool {
+	return i != Pending
+}
+
+// IsSuccess returns true if the status code is a final successfull status code
+func (i StatusCode) IsSuccess() bool {
+	switch i {
+	case Succeeded:
+		return true
+
+	case Queued:
+		return true
+
+	default:
+		return false
+	}
+}
+
+// FailureStatusCode indicates the status of failuted ingestion attempts
+type FailureStatusCode string
+
+const (
+	// Unknown represents an undefined or unset failure state
+	Unknown FailureStatusCode = "Unknown"
+	// Permanent represnets failure state that will benefit from a retry attempt
+	Permanent FailureStatusCode = "Permanent"
+	// Transient represnet a retryable failure state
+	Transient FailureStatusCode = "Transient"
+	// Exhausted represents a retryable failure that has exhusted all retry attempts
+	Exhausted FailureStatusCode = "Exhausted"
+)
+
+// IsRetryable indicates whether there's any merit in retying ingestion
+func (i FailureStatusCode) IsRetryable() bool {
+	switch i {
+	case Transient:
+		return true
+
+	case Exhausted:
+		return true
+
+	default:
+		return false
+	}
+}
+
+// statusRecord is a record containing information regarding the status of an ingation command
+type statusRecord struct {
+	// Status - The ingestion status returned from the service. Status remains 'Pending' during the ingestion process and
+	// is updated by the service once the ingestion completes. When <see cref="IngestionReportMethod"/> is set to 'Queue', the ingestion status
+	// will always be 'Queued' and the caller needs to query the reports queues for ingestion status, as configured. To query statuses that were
+	// reported to queue, see: <see href="https://docs.microsoft.com/en-us/azure/kusto/api/netfx/kusto-ingest-client-status#ingestion-status-in-azure-queue"/>.
+	// When <see cref="IngestionReportMethod"/> is set to 'Table', call <see cref="IKustoIngestionResult.GetIngestionStatusBySourceId"/> or
+	// <see cref="IKustoIngestionResult.GetIngestionStatusCollection"/> to retrieve the most recent ingestion status.
+	Status StatusCode
+
+	// IngestionSourceID - A unique identifier representing the ingested source. It can be supplied during the ingestion execution.
+	IngestionSourceID uuid.UUID
+
+	// The URI of the blob, potentially including the secret needed to access

```

Comment:
Must start with the name of the variable, IngestionSourcePath and complete sentences.